File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.py
def solution(k, tangerine):
    answer = 0
    cnt = 0
    
    dic = {}
    
    for i in tangerine:
        dic[i] = dic.get(i, 0) + 1
            
    dicItems = sorted(dic.values(), reverse=True)
    
    for i in dicItems:
        if (cnt >= k):
            return answer
        
        cnt += i
        answer += 1
    
    return answer

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.java
import java.util.*;

class Solution {
    public int solution(int k, int[] tangerine) {
        int answer = 0;
        
        Map<Integer, Integer> map = new HashMap<>();
        
        for (int i=0; i<tangerine.length; i++) {
            map.put(tangerine[i], map.getOrDefault(tangerine[i], 0) + 1);
        }
        
        Collection<Integer> values = map.values();
        
        // Collection을 List로 변환
        List<Integer> valueList = new ArrayList<>(values);

        // 값 정렬 (내림차순)
        Collections.sort(valueList, Collections.reverseOrder());
        
        for (int i=0; i<valueList.size(); i++) {
            if (k-valueList.get(i) <= 0) {
                answer+=1;
                break;
            }
            k-=valueList.get(i);
            answer+=1;
        }
        
        return answer;
    }
}